app/services/register_service.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Service for registering a new ETD in the system
+class RegisterService
+  ETD_APO_DRUID = 'druid:bx911tp9024' # the APO that governs all ETDs
+
+  # @return [Submission] the newly registered ETD, which is modified and the caller should call save on the return value
+  def self.register(submission:)
+    return submission if submission.druid.present?
+
+    request_model = Cocina::Models.build_request({
+                                                   'type' => Cocina::Models::ObjectType.object,
+                                                   'version' => 1,
+                                                   'label' => submission.title,
+                                                   'administrative' => {
+                                                     'hasAdminPolicy' => ETD_APO_DRUID
+                                                   },
+                                                   'identification' => {
+                                                     'sourceId' => "dissertation:#{submission.dissertation_id}"
+                                                   }
+                                                 })
+    model_response = Dor::Services::Client.objects.register(params: request_model)
+
+    submission.update(druid: model_response.externalIdentifier)
```

Comment:
```suggestion
    submission.update!(druid: model_response.externalIdentifier)
```

Reply:
You can't splat this early to save because the submission does not validate until the next line in the controller with the submission attributes.

Reply:
WFM - Give me 1 minute.